core/domain/acl_decorators.py

```diff
@@ -77,7 +118,8 @@ def test_can_play(self, exploration_id, **kwargs):
                 exploration_id):
             return handler(self, exploration_id, **kwargs)
         else:
-            raise self.PageNotFoundException
+            self.redirect('/')
+            return
```

Comment:
@seanlip ..... When a user tries to play an exploration to which he doesn't have permission, what is better - showing it a) Page Not Found, b) showing it unauthorized or c) redirecting it to home page?